src/tlo/methods/enhanced_lifestyle.py

```diff
@@ -826,533 +455,1438 @@ def on_birth(self, mother_id, child_id):
+                        Steps:
+                            1.  accept a population dataframe from the predict method of a custom linear model
+                            2.  create a new series using `li_no_access_handwashing` column data in the dataframe
+                            3.  update the dataframe using different no access handwashing probabilities
+                            4.  return the updated no access handwashing series
 
-        eff_rate_improved_sanitation = pd.Series(
-            m.parameters['r_improved_sanitation'], index=unimproved_sanitaton_idx
-        )
+                        :param self: a reference to the calling custom linear model
+                        :param df: population dataframe
+                        :param rng: random number generator
+                        :param externals: any external variable passed on to this function """
+            # get module parameters
+            p = self.parameters
 
-        random_draw = rng.random_sample(len(unimproved_sanitaton_idx))
+            # create a new dataframe that will hold all no access handwashing transitions
+            trans_no_access_handwashing = pd.Series(data=df.li_no_access_handwashing.copy(), index=df.index)
 
-        newly_improved_sanitation_idx = unimproved_sanitaton_idx[random_draw < eff_rate_improved_sanitation]
-        df.loc[newly_improved_sanitation_idx, 'li_unimproved_sanitation'] = False
-        df.loc[newly_improved_sanitation_idx, 'li_date_acquire_improved_sanitation'] = self.sim.date
+            # probability of moving to access to handwashing at all follow-up times
+            no_access_handwashing_idx = df.index[df.li_no_access_handwashing & df.is_alive]
 
-        # probability of improved sanitation upon moving to urban from rural
-        unimproved_sanitation_newly_urban_idx = df.index[
-            df.li_unimproved_sanitation & df.is_alive & (df.li_date_trans_to_urban == self.sim.date)
-            ]
+            eff_rate_access_handwashing = pd.Series(p['r_access_handwashing'], index=no_access_handwashing_idx)
 
-        random_draw = rng.random_sample(len(unimproved_sanitation_newly_urban_idx))
+            random_draw = rng.random_sample(len(no_access_handwashing_idx))
 
-        eff_prev_unimproved_sanitation_urban = pd.Series(
-            m.parameters['init_p_unimproved_sanitation_urban'], index=unimproved_sanitation_newly_urban_idx
-        )
+            newly_access_handwashing_idx = no_access_handwashing_idx[random_draw < eff_rate_access_handwashing]
+            trans_no_access_handwashing.loc[newly_access_handwashing_idx] = False
 
-        df.loc[unimproved_sanitation_newly_urban_idx, 'li_unimproved_sanitation'] = (
-            random_draw < eff_prev_unimproved_sanitation_urban
-        )
+            # setting `li_date_acquire_access_handwashing` directly here, perhaps I can do better?
+            df.loc[newly_access_handwashing_idx, 'li_date_acquire_access_handwashing'] = externals['other']
```

Comment:
I think it would be better to do this outside the linear model predict function as generally the expectation is that the linear model doesn't update the population dataframe itself.